File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;
        int maj,cnt;
        cnt = 0;
        maj = 0;

        for(int ele : nums){
            if(cnt == 0){
                maj = ele;
                cnt = 1;
            }else if(ele == maj){
                cnt++;
            }else{
                cnt--;
            }
        }
        return maj;
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        d={}
        for i in set(nums):
            d[i]=nums.count(i)
        for i in d.keys():
            if(d[i]>len(nums)/2):
                return i
        return 0;
        